core/src/config.jl

```diff
@@ -233,19 +233,26 @@ end
 
 "Convert the saveat Float64 from our Config to SciML's saveat"
 function convert_saveat(saveat::Float64, t_end::Float64)::Union{Float64, Vector{Float64}}
+    errors = false
     if iszero(saveat)
         # every step
-        Float64[]
+        saveat = Float64[]
     elseif saveat == Inf
         # only the start and end
-        [0.0, t_end]
+        saveat = [0.0, t_end]
```

Comment:
Would it not be simpler to return `t_end` (or perhaps `[t_end]`) here instead of filtering the 0.0 out for the SavingCallback?

Reply:
I don't know whether that's desirable since `saveat` is also used for the save subgrid level callback.

Reply:
Yeah I was thinking for that we can use save start = true